src/notebooks/outputs/tracebackFormatter.unit.test.ts

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { assert } from 'chai';
+import { instance, mock, when } from 'ts-mockito';
+import { NotebookCell, NotebookDocument, TextDocument, Uri } from 'vscode';
+import { JupyterNotebookView } from '../../platform/common/constants';
+import { NotebookTracebackFormatter } from './tracebackFormatter';
+
+suite(`Notebook trace formatter`, function () {
+    let notebook: NotebookDocument;
+    let cell: NotebookCell;
+    let document: TextDocument;
+
+    setup(() => {
+        notebook = mock<NotebookDocument>();
+        when(notebook.notebookType).thenReturn(JupyterNotebookView);
+        cell = mock<NotebookCell>();
+        when(cell.index).thenReturn(0);
+        when(cell.notebook).thenReturn(instance(notebook));
+        document = mock<TextDocument>();
+        const uri = Uri.parse('vscode-notebook-cell:/Users/USR/notebooks/error.ipynb#W0sZmlsZQ==');
+        when(document.uri).thenReturn(uri);
+        when(cell.document).thenReturn(instance(document));
+    });
+
+    test('ipython: 8.3.0, ipykernel: 6.13.0', function () {
+        const formatter = new NotebookTracebackFormatter();
+        const traceback = [
```

Comment:
can you put in a note about how to generate this input? I remember this being a difficult part last time I worked on it